weave-js/src/components/PagePanelComponents/Home/Browse3/pages/common/EmptyContent.tsx

```diff
@@ -1,14 +1,84 @@
 import {Box} from '@mui/material';
 import React, {useState} from 'react';
+import {useParams} from 'react-router-dom';
+import {toast} from 'react-toastify';
 
 import {TargetBlank} from '../../../../../../common/util/links';
 import {Button} from '../../../../../Button';
+import {useWeaveflowCurrentRouteContext} from '../../context';
 import {CreateDatasetDrawer} from '../../datasets/CreateDatasetDrawer';
+import {createNewDataset} from '../../datasets/datasetOperations';
+import {DatasetPublishToast} from '../../datasets/DatasetPublishToast';
+import {useWFHooks} from '../../pages/wfReactInterface/context';
 import {EmptyProps} from './Empty';
 import {Link} from './Links';
 
 const NewDatasetButton: React.FC = () => {
   const [isDrawerOpen, setIsDrawerOpen] = useState(false);
+  const [isCreatingDataset, setIsCreatingDataset] = useState(false);
+  const {entity, project} = useParams<{entity: string; project: string}>();
+  const router = useWeaveflowCurrentRouteContext();
+  const {useObjCreate, useTableCreate} = useWFHooks();
+
+  // Get the create hooks
+  const tableCreate = useTableCreate();
+  const objCreate = useObjCreate();
+
+  const handleSaveDataset = async (dataset: any) => {
+    // Check if this is a publish action
+    const isPublish = dataset.publishNow === true;
+
+    setIsCreatingDataset(true);
+    try {
+      // Parse the rows from string back to array if they are provided as a string
```

Comment:
when are they a string vs array? maybe comment this, seems like a pretty different flow